cli/cmd/list_cmd.go

```diff
@@ -2,55 +2,56 @@ package cmd
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
 	"github.com/kubeshop/tracetest/cli/parameters"
 	"github.com/kubeshop/tracetest/cli/pkg/resourcemanager"
 	"github.com/spf13/cobra"
 )
 
-var listParams = parameters.ListParams{}
-
-var listCmd = &cobra.Command{
-	GroupID: cmdGroupResources.ID,
-	Use:     fmt.Sprintf("list %s", strings.Join(parameters.ValidResources, "|")),
-	Short:   "List resources",
-	Long:    "List resources from your Tracetest server",
-	PreRun:  setupCommand(),
-	Run: WithResourceMiddleware(func(_ *cobra.Command, args []string) (string, error) {
-		resourceType := args[0]
-		ctx := context.Background()
-
-		resourceClient, err := resources.Get(resourceType)
-		if err != nil {
-			return "", err
-		}
-
-		resultFormat, err := resourcemanager.Formats.Get(output, "pretty")
-		if err != nil {
-			return "", err
-		}
-
-		lp := resourcemanager.ListOption{
-			Take:          listParams.Take,
-			Skip:          listParams.Skip,
-			SortBy:        listParams.SortBy,
-			SortDirection: listParams.SortDirection,
-			All:           listParams.All,
-		}
-
-		result, err := resourceClient.List(ctx, lp, resultFormat)
-		if err != nil {
-			return "", err
-		}
-
-		return result, nil
-	}, listParams),
-	PostRun: teardownCommand,
-}
+var (
+	listParams = parameters.ListParams{}
+	listCmd    *cobra.Command
+)
 
 func init() {
+	listCmd = &cobra.Command{
+		GroupID: cmdGroupResources.ID,
+		Use:     "list " + resourceList(),
+		Short:   "List resources",
+		Long:    "List resources from your Tracetest server",
+		PreRun:  setupCommand(),
+		Run: WithResourceMiddleware(func(_ *cobra.Command, args []string) (string, error) {
+			resourceType := args[0]
+			ctx := context.Background()
```

Comment:
Same question as I did on `get`. Can we have a panic here if the user executes only `tracetest list`?

Reply:
same answer as on `get`. this ges validated on the middleware before getting here